src/script/interpreter.cpp

```diff
@@ -1199,18 +1199,14 @@ class CScCertificateSignatureSerializer {
         std::vector<CTxOut> vout_ser;
 
         // reading from memory and writing to data stream
-        for (auto it = certTo.GetVout().begin(); it != certTo.GetVout().end(); ++it)
+        for(int pos = 0; pos < certTo.GetVout().size(); ++pos)
```

Comment:
CScCertificateSignatureSerializer does not serialize properly inputs taking in consideration selected nIn. Moreover there is no test verifying the correct behavior of certificate's signature! 
Remove CScCertificateSignatureSerializer class and "adapt" CTransactionSignatureSerializer to manage CTransactionBase and just differentiate the behavior in the Serialize method to keep the previous logic for tx and instead for certs serialize as in CCertficate  (only differences should be LockTime, Joinsplits, Bwts).
Introduce a test passing a Certificate instance to CTransactionSignatureSerializer.